glossary.yml

```
@@ -1749,6 +1749,22 @@
     def: >
       An unambiguous name of the form `package::thing`.
 
+- slug: function
+  ref:
+    - functions
+    - command
+  en:
+    term: "function"
+    def: >
+      A code block which gathers a sequence of operations into a whole, preserving it for ongoing use
+      by a defining tasts that take a set of inputs and generate expected outputs. They enable
+      repeating these defined tasks with one command.
+  ja:
+    term: "関数"
+    def: >
+      連続した操作を一つに纏めようにコードブロックとして後から使うときのために保存しておきます。
+      関数は繰り繰返し演算を実行できるのように入力と返した出力を定義します。
```

Agreed with naoe-tatara:
```suggestion
      関数は繰り繰返し演算を実行できるように入力と返した出力を定義します。
```

---

I agree to this change. I think I got confused with the Kanji for "iterative" (operation).

```diff
-関数は繰り繰返し演算を実行できるように入力と返した出力を定義します。
+関数は繰り返し演算を実行できるように入力と返した出力を定義します。
```

I'm not sure about the other point. I haven't mentioned 引数 for "argument" (also called option, flag, or parameter), although I think this would a correct translation in Japanese. I actually left this out of the English definition as well for simplicity and consistency. 

I'm trying to be consistent with the translated R lessons here:

https://swcarpentry-ja.github.io/r-novice-gapminder/10-functions/index.html
https://swcarpentry-ja.github.io/r-novice-gapminder/ja/_episodes/10-functions/index.html

Here we have the following definition where "出力" refers to the output. So my intention here is "返した出力" as "the output(s) that is/are returned".
 * a defined set of inputs and expected outputs
* 入力と出力の組み合わせを定義

Unfortunately we haven't translated the relevant section for Shell or this would be helpful.

https://swcarpentry-ja.github.io/shell-novice/01-intro/index.html

---

I've added a "optional arguments" to both definitions to clarify this.

---

Thanks for the clarification. What about just '出力', or '返された出力' or 'リターンされた出力'? It could be just me feeling 返した出力 does not sound right, so, I need a third opinion - What do you think, @naoe-tatara ?

---

Ah I see. Passive form makes sense to me.  Of course, as a non-native speaker it is good for me to know if it's unnatural Japanese. Wikipedia for [サブルーチン](https://ja.wikipedia.org/wiki/%E3%82%B5%E3%83%96%E3%83%AB%E3%83%BC%E3%83%81%E3%83%B3#%E5%91%BC%E7%A7%B0) / [Function](https://en.wikipedia.org/wiki/Function_(computer_science)) uses 返り値 for "returned value".

---

Updated for consistency with English version:

* returns expected outputs (return values), if any.
* もしあれば返された出力(返り値)を定義します。

---

Thanks @TomKellyGenetics !